src/actions/validators.js

```diff
@@ -0,0 +1,27 @@
+import { Jurisdiction } from '../contracts'
+import FetchingActions from './fetching'
+import * as ActionTypes from '../actiontypes'
+import { JURISDICTION_ADDRESS } from '../constants'
+import Network from "../network";
+
+const ValidatorsActions = {
+
+  findAll() {
+    return async function(dispatch) {
+      const jurisdiction = await Jurisdiction.at(JURISDICTION_ADDRESS)
+      const events = jurisdiction.ValidatorAdded({}, { fromBlock: 0, toBlock: 'latest' });
+      events.watch(function(error, result) {
+        if(error) throw error
+        else dispatch(ValidatorsActions.add(result.args.validator))
+      })
+    }
+  },
+
+  add(validator) {
```

Comment:
This can be a synchronous action. (Notice how it's always immediately dispatched.)

```javascript
add(validator) {
  return { type: ActionTypes.ADD_VALIDATOR, validator }
}
```

Reply:
right, updated.